tests/api/test_pagination.py

```diff
@@ -0,0 +1,1654 @@
+            ),
+        ]
+    )
+
+
+@pytest.fixture
+def pages_for_pagination():
+    Page.objects.bulk_create(
+        [
+            Page(title="About", slug="about", is_published=False),
+            Page(title="About", slug="about2", is_published=False),
+            Page(title="About3", slug="about3", is_published=False),
+            Page(title="Page1", slug="slug_page_1", is_published=True),
+            Page(title="Page2", slug="slug_page_2", is_published=True),
+        ]
+    )
+
+
+@pytest.fixture
+def attributes_for_pagination():
+    return Attribute.objects.bulk_create(
+        [
+            Attribute(
+                name="A",
+                slug="a",
+                value_required=True,
+                is_variant_only=False,
+                visible_in_storefront=False,
+                filterable_in_storefront=True,
+                filterable_in_dashboard=True,
+                available_in_grid=True,
+                storefront_search_position=2,
+            ),
+            Attribute(
+                name="A",
+                slug="b",
+                value_required=True,
+                is_variant_only=True,
+                visible_in_storefront=True,
+                filterable_in_storefront=False,
+                filterable_in_dashboard=False,
+                available_in_grid=True,
+                storefront_search_position=5,
+            ),
+            Attribute(
+                name="A",
+                slug="c",
+                value_required=False,
+                is_variant_only=False,
+                visible_in_storefront=False,
+                filterable_in_storefront=False,
+                filterable_in_dashboard=True,
+                available_in_grid=True,
+                storefront_search_position=2,
+            ),
+            Attribute(
+                name="B",
+                slug="d",
+                value_required=False,
+                is_variant_only=True,
+                visible_in_storefront=True,
+                filterable_in_storefront=True,
+                filterable_in_dashboard=True,
+                available_in_grid=False,
+                storefront_search_position=4,
+            ),
+            Attribute(
+                name="C",
+                slug="e",
+                value_required=False,
+                is_variant_only=False,
+                visible_in_storefront=True,
+                filterable_in_storefront=True,
+                filterable_in_dashboard=False,
+                available_in_grid=False,
+                storefront_search_position=5,
+            ),
+        ]
+    )
+
+
+@pytest.fixture
+def categories_for_pagination():
+    product_types = ProductType.objects.bulk_create(
+        [
+            ProductType(name="Default Type", slug="default-type",),
+            ProductType(name="Default Type", slug="second-type",),
+        ]
+    )
+
+    category1 = Category.objects.create(name="The Category", slug="the-category",)
```

Comment:
Bulk?

Reply:
category below use this one as parent
using bulk I would have to update cetegories below

Reply:
https://github.com/mirumee/saleor/pull/5149/files/82318b7f6a5eb6f34ffacb088336a525adaeda87#diff-f177a26d6e2119236d7f092903f38899R408-R409
This can be done.